docs/references/strategies/languages/python/pipenv.md

```diff
@@ -0,0 +1,339 @@
+# Pipenv/Pipfile
+
+This is a modern approach to defining python project dependencies, providing
+very precise, complete dependency graphs for a python project.
+
+## Project Discovery
+
+Find all files named `Pipfile.lock`
+
+## Analysis
+
+We parse `Pipfile.lock` -- a json-structured file -- to find:
+
+- `sources` - repositories/locations that can be referenced by packages
+- `default` - production dependencies
+- `develop` - development dependencies
+
+Dependencies may contain an `index` field -- this is a reference to a repository
+in the top-level `sources`.
+
+Dependencies contain pinned `version` information.
+
+When possible, we use `pipenv graph --json-tree` to hydrate the edges between
+dependencies. This will fail unless `pipenv install` has been run in that directory.
+
+## Limitations
+
+- Pipfile.lock does not report edges, if `pipenv graph --json-tree` fails, we will
+not be able to detect the edges between dependencies.
+
+  - `pipenv graph --json-tree` relies on `pipenv install` being run in the directory beforehand.
+  If that command was not run (such as in a freshly cloned repo), then the graph command will fail,
+  which means that pipenv analysis will succeed, but with incomplete data (missing edges)
```

Comment:
I think this document largely predated this PR, but I think something that would be useful to add as a documentation guideline is to clarify to users what issues concretely mean for them when we talk about the issue.

For example, if I'm a user the result of "missing edges" is unclear. I think it means "all the dependencies will appear as direct in the UI"; if this is correct that is probably useful information to provide (either inline, or as a link to a troubleshooting document where we explain the behavior and underlying issue).